src/category_theory/limits/concrete_category.lean

```diff
@@ -24,30 +25,80 @@ local attribute [instance] concrete_category.has_coe_to_fun
 -- We now prove a lemma about naturality of cones over functors into bundled categories.
 namespace cone
 
-variables {J : Type u} [small_category J]
-
 /-- Naturality of a cone over functors to a concrete category. -/
-@[simp] lemma naturality_concrete {G : J ⥤ C} (s : cone G) {j j' : J} (f : j ⟶ j') (x : s.X) :
-   (G.map f) ((s.π.app j) x) = (s.π.app j') x :=
+@[simp] lemma w_apply {F : J ⥤ C} (s : cone F) {j j' : J} (f : j ⟶ j') (x : s.X) :
+   F.map f (s.π.app j x) = s.π.app j' x :=
 begin
-  convert congr_fun (congr_arg (λ k : s.X ⟶ G.obj j', (k : s.X → G.obj j')) (s.π.naturality f).symm) x;
-  { dsimp, simp [-cone.w] },
+  convert congr_fun (congr_arg (λ k : s.X ⟶ F.obj j', (k : s.X → F.obj j')) (s.w f)) x,
+  simp only [coe_comp],
 end
 
+@[simp]
+lemma w_forget_apply (F : J ⥤ C) (s : cone (F ⋙ forget C)) {j j' : J} (f : j ⟶ j') (x : s.X) :
+  F.map f (s.π.app j x) = s.π.app j' x :=
+congr_fun (s.w f : _) x
+
 end cone
 
 namespace cocone
 
-variables {J : Type u} [small_category J]
-
 /-- Naturality of a cocone over functors into a concrete category. -/
-@[simp] lemma naturality_concrete {G : J ⥤ C} (s : cocone G) {j j' : J} (f : j ⟶ j') (x : G.obj j) :
-  (s.ι.app j') ((G.map f) x) = (s.ι.app j) x :=
+@[simp] lemma w_apply {F : J ⥤ C} (s : cocone F) {j j' : J} (f : j ⟶ j') (x : F.obj j) :
+  s.ι.app j' (F.map f x) = s.ι.app j x :=
 begin
-  convert congr_fun (congr_arg (λ k : G.obj j ⟶ s.X, (k : G.obj j → s.X)) (s.ι.naturality f)) x;
-  { dsimp, simp [-nat_trans.naturality, -cocone.w] },
+  convert congr_fun (congr_arg (λ k : F.obj j ⟶ s.X, (k : F.obj j → s.X)) (s.w f)) x,
+  simp only [coe_comp],
 end
 
+@[simp]
+lemma w_forget_apply (F : J ⥤ C) (s : cocone (F ⋙ forget C)) {j j' : J} (f : j ⟶ j') (x : F.obj j) :
+  s.ι.app j' (F.map f x) = (s.ι.app j x : s.X) :=
+congr_fun (s.w f : _) x
+
 end cocone
 
+section has_limit
+
+@[simp]
+lemma limit.lift_π_apply (F : J ⥤ C) [has_limit F] (s : cone F) (j : J) (x : s.X) :
+  limit.π F j (limit.lift F s x) = s.π.app j x :=
+begin
+  have w := congr_arg (λ f : s.X ⟶ F.obj j, (f : s.X → F.obj j) x) (limit.lift_π s j),
+  dsimp at w,
+  rwa coe_comp at w,
+end
+
+@[simp]
+lemma limit.w_apply (F : J ⥤ C) [has_limit F] {j j' : J} (f : j ⟶ j') (x : limit F) :
```

Comment:
Why this name? There is no `w` in the statement.

Reply:
see above